roles/cloudfront/tasks/destroy.yml

```diff
@@ -0,0 +1,25 @@
+---
+
+- name: Retrieve our cloudfront distro if it exists
+  cloudfront_facts:
+    distribution: true
+    domain_name_alias: "{{ item.alias_domains.0 }}"
+  register: cf_facts
+
+- name: Set fact if we have have a cloudfront distribution
+  set_fact:
+    cf_exists: "{{ cf_facts['ansible_facts']['cloudfront'][item.alias_domains.0]['Distribution']['Id'] is defined }}"
+
+- block:
+    # Possibly fire and forget this, there are no sg's attached and once we remove the A
+    # records it is inacessible
```

Comment:
I would leave that wait_for_deployed set as it is now, it means that we know where we stand by the time this play finishes